torch/_refs/nn/functional/__init__.py

```diff
@@ -45,3 +80,78 @@ def elu(
         rhs = refs.expm1(a)
 
     return refs.where(refs.gt(a, 0), a, rhs)
+
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.OP_MATH,
+)
+def selu(
+    a: TensorLikeType, inplace: bool = False
+) -> TensorLikeType:
+    """
+    Reference implementation of torch.nn.functional.selu
+    """
+
+    if inplace:
+        raise NotImplementedError
+
+    alpha = 1.6732632423543772848170429916717
+    scale = 1.0507009873554804934193349852946
+
+    python_type = utils.dtype_to_type(a.dtype)
+    if not utils.is_weakly_lesser_type(type(alpha), python_type):
+        msg = (
+            "alpha constant of type {0} cannot be safely cast to type {1}!".format(
+                type(alpha), python_type
+            )
+        )
+        raise ValueError(msg)
+    if not utils.is_weakly_lesser_type(type(scale), python_type):
+        msg = (
+            "scale constant of type {0} cannot be safely cast to type {1}!".format(
+                type(scale), python_type
+            )
+        )
+        raise ValueError(msg)
+
+    rhs = refs.mul(alpha, refs.expm1(a))
+
+    # FIXME: If I put the mul on the outside, I hide a stride consistency error!
+    # return refs.mul(scale, refs.where(refs.gt(a, 0), a, rhs))
+    return refs.where(refs.gt(a, 0), refs.mul(scale, a), refs.mul(scale, rhs))
+
+# softplus is implemented specially because it has beta and threshold arguments
+@elementwise_type_promotion_wrapper(
+    type_promoting_args=("a",),
+    type_promotion_kind=ELEMENTWISE_TYPE_PROMOTION_KIND.OP_MATH,
+)
+def softplus(
```

Comment:
I don't totally understand the current test failures, because they seem to be complaining about torch.nn.functional.softplus, and not refs.softplus - maybe we have a bug in our tests?

However, they are hinting at a legitimate issues, which is that softplus actually has a secret out= parameter:

https://github.com/pytorch/pytorch/blob/dcd2ba353844bbc7d480bf3aa15fd0eeb0d55e78/aten/src/ATen/native/native_functions.yaml#L9566

So this probably needs to apply the out_wrapper from _/prims/wrappers.py

Reply:
I think this was my fault as I added `supports_out=False` to the `softplus` `opInfo` entry because I was getting the following error when testing `_refs.softplus`.  

```
======================================================================
ERROR: test_out__refs_nn_functional_softplus_cpu_float32 (__main__.TestCommonCPU)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/workspace/pytorch/torch/testing/_internal/common_device_type.py", line 377, in instantiated_test
    result = test(self, **param_kwargs)
  File "/workspace/pytorch/torch/testing/_internal/common_device_type.py", line 786, in test_wrapper
    return test(*args, **kwargs)
  File "test/test_ops.py", line 738, in test_out
    _compare_out(_case_zero_transform)
  File "test/test_ops.py", line 713, in _compare_out
    op_out(out=out)
  File "/workspace/pytorch/torch/testing/_internal/common_methods_invocations.py", line 930, in __call__
    return self.op(*args, **kwargs)
  File "/workspace/pytorch/torch/_prims/wrappers.py", line 95, in _fn
    bound = sig.bind(*args, **kwargs)
  File "/opt/conda/lib/python3.8/inspect.py", line 3037, in bind
    return self._bind(args, kwargs)
  File "/opt/conda/lib/python3.8/inspect.py", line 3026, in _bind
    raise TypeError(
TypeError: got an unexpected keyword argument 'out'
```